specs/latest/2.0/index.html

```diff
@@ -1869,7 +1869,7 @@ <h2 class="no-toc">Table of contents</h2>
         </dt>
         <dd>
           <p>If no WebGLBuffer is bound to the <code>PIXEL_PACK_BUFFER</code> target, generates an <code>INVALID_OPERATION</code> error.</p>
-          <p><em>offset</em> is the byte offset into the WebGLBuffer's data store; generates an <code>INVALID_VALUE</code> error if it's less than 0.</p>
+          <p><em>offset</em> is the byte offset into the WebGLBuffer's data store; generates an <code>INVALID_VALUE</code> error if it's less than 0. If <em>offset</em> passes the end of the pixel pack buffer, or the pixel pack buffer is not large enough to store all pixels, generates an <code>INVALID_OPERATION</code>.</p>
```

Comment:
bufferSubData generates INVALID_VALUE (not INVALID_ENUM or INVALID_OPERATION) if there is an attempt to write data past the end of the buffer's data store.

I think readPixels should also generate INVALID_VALUE if there is an attempt to read back pixel data past the end of the bound pixel pack buffer.